src/goto-programs/read_goto_binary.cpp

```diff
@@ -323,7 +327,10 @@ static bool link_functions(
 
   if(!macro_application.expr_map.empty())
     Forall_goto_functions(dest_it, dest_functions)
-      rename_symbols_in_function(dest_it->second, macro_application);
+    {
+      irep_idt final_id=dest_it->first;
+      rename_symbols_in_function(dest_it->second, final_id, macro_application);
+    }
```

Comment:
I don't think this change is necessary, the cases of renamed functions should all be caught above.